### PR TITLE
#176168716 chore(context): setup context for state management

### PR DESCRIPTION
**Pull request by SimonImanigirimpuhwe**
#### What does this PR do?

- Setup context for state management
-  configure toast

#### Description of Task to be completed? 

- Setup context

- Add authentication to test context

- Add profile for logged user

- Display toast when authentication succeed or failed

#### How should this be manually tested?

- Clone the repo 

- In your home directory  cd this branch **`ch-setup-context-#176168716`**

- Then run **`npm install`** to install all dependencies

- Run **`npm start`** to start the server

- Run **`npm test`** to run tests

- In your default browser navigate to **`http://localhost:3000/login`**

> When accurate credentials provided you should see a toast for the action and profile display

#### Any background context you want to provide?

- To be able to login successfully you should login with the following credentials
```
 Email: 'johndoe@gmail.com'
 Password: 'JohnDoe123'
```

#### What are the relevant Pivotal Tracker stories?

- [#176168716](https://www.pivotaltracker.com/n/projects/2476677/stories/176168716)

#### Screenshots (if appropriate)
Screen Shot 2020-12-16 at 10 37 47

#### Questions:

- N/A

### CODE REVIEW

File: src/contexts/AuthContext.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+export const AuthContext = createContext();
+
+const AuthContextProvider = (props) => {
+    const [auth, dispatch] = useReducer(authReducer, {error: null, isAuthenticated: false})
+    const userInfo = {
+        Name:'John Doe',
+        Email: 'johndoe@gmail.com',
+        Password: 'JohnDoe123'
+    }
```
Comment (malaba6) [inline src/contexts/AuthContext.js:14]: Plz include the user avatar aswell in the user context

File: src/helpers/toasts.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        return toast.info(message)
+    } else if (type === 'warn') {
+        return toast.warn(message)
+    } else if (type === 'error') {
+        return toast.error(message)
+    } else {
+        return toast(message)
+    }
+};
+console.log(toast)
```
Comment (malaba6) [inline src/helpers/toasts.js]: Plz remove this console.log line

File: src/views/Login/Login.js [hunk 3]
```diff
@@ -35,8 +43,41 @@ export default function SignIn() {
...
+          toaster(auth.error, 'error')
+          return false
+        }
+         else if (values.password !== savedUser.Password) {
+          dispatch({type: SET_ERROR, payload: 'invalid Password'})
+          toaster(auth.error, 'error')
+          return false
+        }
+          dispatch({type: SET_AUTHENTICATION})
+          toaster('You are logged in', 'succes')
```
Comment (malaba6) [inline src/views/Login/Login.js]: Typo here, success not succes

File: src/views/Login/Login.js [hunk 4]
```diff
@@ -35,8 +43,41 @@ export default function SignIn() {
...
+          console.log('logged')
```
Comment (malaba6) [inline src/views/Login/Login.js]: Remove this console line

**Review by malaba6 — CHANGES_REQUESTED** [review]

**Review by thepatrickniyo — APPROVED** [review]

**Review by SamuelSabasNayo — APPROVED** [review]

**Review by joshwambere — COMMENTED** [review]
When you input the wrong credentials it exactly tells you what you got wrong, eg: if you enter the wrong password it tells you provided the wrong password I think for security reasons it shouldn't

**Pull request merged**